### ISSUE TITLE
torch.empty() bug

### ISSUE THREAD

**Issue by chuxiang93** [NVIDIA/apex#1583]
https://github.com/NVIDIA/apex/blob/ba027dd0bac621a5c3d16bfb90d73e2d48d2588e/apex/amp/compat.py#L9

The torch empty function must take a size parameter.

**Comment by MozerWang** [comment]
BUG! I met the same problem!

**Comment by LiuXinqi12** [comment]
BUG! I met the same problem!

**Comment by xmfbit** [comment]
Yes, see https://github.com/NVIDIA/apex/commit/ba027dd0bac621a5c3d16bfb90d73e2d48d2588e#r100574972

cc @NouamaneTazi  Could you verify this?

**Comment by LiuXinqi12** [comment]
Yes, replacing "x = torch.empty()" to  "x = torch.Tensor()" in  apex/amp/compat.py can solve this issue.

**Comment by crcrpar** [comment]
https://github.com/NVIDIA/apex/pull/1584 reverted the change

**Comment by crcrpar** [comment]
note that amp has been deprecated in favor of pytorch `autocast` and GradScaler`. also we're going to remove amp from the repo

**Comment by xmfbit** [comment]
> Yes, see [ba027dd#r100574972](https://github.com/NVIDIA/apex/commit/ba027dd0bac621a5c3d16bfb90d73e2d48d2588e#r100574972)
> 
> cc @NouamaneTazi Could you verify this?

@crcrpar It seems that this problem has not been fixed. I have posted in the link how to reproduce the bug.

**Comment by xmfbit** [comment]
> note that amp has been deprecated in favor of pytorch `autocast` and GradScaler`. also we're going to remove amp from the repo

What about the `contrib/sparsity` package?
